tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -45,18 +45,21 @@ MakeClusters(const std::string &treeName, const std::vector<std::string> &fileNa
    const auto nFileNames = fileNames.size();
    Long64_t offset = 0ll;
    for (auto i = 0u; i < nFileNames; ++i) {
-      std::unique_ptr<TFile> f(TFile::Open(fileNames[i].c_str())); // need TFile::Open to load plugins if need be
-      TTree *t = nullptr; // not a leak, t will be deleted by f
-      f->GetObject(treeName.c_str(), t);
-      auto clusterIter = t->GetClusterIterator(0);
-      Long64_t start = 0ll, end = 0ll;
-      const Long64_t entries = t->GetEntries();
-      // Iterate over the clusters in the current file
+      Long64_t entries = 0L;
       std::vector<EntryCluster> clusters;
-      while ((start = clusterIter()) < entries) {
-         end = clusterIter.GetNextEntry();
-         // Add the current file's offset to start and end to make them (chain) global
-         clusters.emplace_back(EntryCluster{start + offset, end + offset});
+      std::unique_ptr<TFile> f(TFile::Open(fileNames[i].c_str())); // need TFile::Open to load plugins if need be
+      if (f) { // Suppose there is an issue with the file: we skip it.
```

Comment:
Shouldn't it be
```
   if (f && !f->IsZombie())
```
?

Reply:
This can be added. I just thought the plugin system did return a nullptr in case something goes wrong in the opening procedure (it least this is what happens in Olli's example)

Reply:
yes, but sometimes (corrupted file), it might still return a Zombie ...

Reply:
or more exactly, I can not guaranteed there no code path where it could .. and the additional test is cheap.

Reply:
Try with `new TFile("medontexist.root")`...